examples/platform/linux/AppMain.cpp

```diff
@@ -403,10 +404,12 @@ static uint16_t WiFiPAFGet_FreqList(const char * args, std::unique_ptr<uint16_t[
         freq_vect.push_back(std::stoi(item));
     }
     uint16_t freq_size = freq_vect.size();
-    freq_list          = std::make_unique<uint16_t[]>(freq_size);
+    CHIP_ERROR err     = knownFreqListBuilder.EnsureAppendCapacity(freq_size);
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, 0);
     for (int i = 0; i < freq_size; i++)
     {
-        freq_list.get()[i] = freq_vect[i];
+        err = knownFreqListBuilder.Append(freq_vect[i]);
+        VerifyOrReturnValue(err == CHIP_NO_ERROR, i);
```

Comment:
Why do we need to return a size when the buffer `TakeBuffer` would contain the underlying size? How about making this return a CHIP_ERROR and be:

```cpp
ReturnErrorOnFailure(knownFreqListBuilder.AppendElements({freq_vect.data(), freq_vect.size()});
```

Would not hold the PR for this though ... it just feels odd to allocate a vector and then re-allocate and have no error handling (error codes and locations are lost without logging).